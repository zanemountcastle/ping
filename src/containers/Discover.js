import React, { Component } from 'react';
import { StyleSheet, Text, View ,  ScrollView, Alert} from 'react-native';
import { Header } from 'react-native-elements';

import { connect } from 'react-redux';

import FetchDiscoveryData from '../actions/FetchDiscoveryData';

import { AppletsFeed, AppletPreview } from '../components';

export class Discover extends Component {
  componentWillMount() {
    this.props.FetchDiscoveryData();
  }

  render() {
      return (
        <View style={styles.container}>
          <Header
            centerComponent={{ text: 'Discover', style: { color: '#4D4C4C', fontSize: 20, fontWeight: '700' } }}
            outerContainerStyles={{ backgroundColor: '#fff' }}
          />
          <AppletsFeed
            feed={this.props.discovery.data}
            fetchApplets={this.props.FetchDiscoveryData}
            isFetching={this.props.discovery.isFetching}
          />
        </View>
      );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    flexDirection: 'column',
  },
});

function mapStateToProps(state) {
  return {
    discovery: state.discovery
  }
};

export default connect(mapStateToProps, { FetchDiscoveryData })(Discover);
