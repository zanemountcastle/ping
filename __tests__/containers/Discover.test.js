import React from 'react';
import Discover from '../../src/containers/Discover';

import ShallowRenderer from 'react-test-renderer/shallow';

describe('discover screen', () => {

  it('renders without crashing', () => {
    // Shallow render <App /> and not it's child components
    // Standard react-test-renderer doesn't allow shallow rendering
    const renderer = new ShallowRenderer();
    expect(renderer.render(<Discover />)).toBeTruthy();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    expect(renderer.render(<Discover />)).toMatchSnapshot();
  });

});
