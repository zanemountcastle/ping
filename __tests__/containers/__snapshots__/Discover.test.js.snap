// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`discover screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Header
    centerComponent={
      Object {
        "style": Object {
          "color": "#4D4C4C",
          "fontSize": 20,
          "fontWeight": "700",
        },
        "text": "Discover",
      }
    }
    outerContainerStyles={
      Object {
        "backgroundColor": "#fff",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Discover page!
    </Text>
  </View>
</View>
`;
